langchain4j-mongodb/src/main/java/dev/langchain4j/store/embedding/mongodb/document/EmbeddingDocument.java

```diff
@@ -0,0 +1,21 @@
+package dev.langchain4j.store.embedding.mongodb.document;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.bson.codecs.pojo.annotations.BsonId;
+
+import java.util.List;
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class EmbeddingDocument {
```

Comment:
it would be nice to move EmbeddingDocument and EmbeddingMatchDocument one level higher and make them package-private, to not expose the internal implementation to the library users

Reply:
It's not possible to make this class package scope protected.. as the MongoDB driver requires a class to be public (to deserialise/serialize from or to  json/bson) 

I try to use a complete custom CodecRegistry, but without success. the only option left is to use a Map. but that makes the code difficult to read/maintain
The only option